plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/columns-control.tsx

```diff
@@ -2,89 +2,164 @@
 	__experimentalToolsPanelItem as ToolsPanelItem,
+	// @ts-expect-error Using experimental features
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	// @ts-expect-error Using experimental features
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+	// @ts-expect-error Using experimental features
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalUnitControl as UnitControl,
+	RangeControl,
 } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
-import { DisplayLayoutControlProps } from '../../types';
-import { getDefaultDisplayLayout } from '../../constants';
+import {
+	LayoutOptions,
+	ProductCollectionLayoutGrid,
+	TemplateLayoutControlProps,
+} from '../../types';
+import {
+	DEFAULT_LAYOUT_GRID_COLUMNS,
+	DEFAULT_LAYOUT_GRID_MINIMUM_COLUMN_WIDTH,
+} from '../../constants';
 
+const autoColumnsLabel = __( 'Column Mode', 'woocommerce' );
+const autoModeLabel = __( 'Auto', 'woocommerce' );
+const manualModeLabel = __( 'Manual', 'woocommerce' );
+const minimumColumnWidthLabel = __( 'Minimum Column Width', 'woocommerce' );
 const columnsLabel = __( 'Columns', 'woocommerce' );
-const toggleLabel = __( 'Responsive', 'woocommerce' );
-const toggleHelp = __(
-	'Automatically adjust the number of columns to better fit smaller screens.',
-	'woocommerce'
-);
 
-const ColumnsControl = ( props: DisplayLayoutControlProps ) => {
-	const { type, columns, shrinkColumns } = props.displayLayout;
-	const showColumnsControl = type === 'flex';
+const ColumnsControl = ( props: TemplateLayoutControlProps ) => {
+	const templateLayout = props.templateLayout as ProductCollectionLayoutGrid;
 
-	const defaultLayout = getDefaultDisplayLayout();
+	// Rather than serializing whether or not they're using
+	// automatic columns we can just check the current
+	// layout. When they're using a minimum width we
+	// are in auto mode, and if nothing is set, we
+	// will just default to auto mode.
+	const [ autoColumns, setAutoColumns ] = useState(
+		templateLayout.minimumColumnWidth !== undefined ||
+			templateLayout.columnCount === undefined
+	);
 
-	const onShrinkColumnsToggleChange = ( value: boolean ) => {
+	const setTemplateLayout = (
+		newTemplateLayout: Partial< ProductCollectionLayoutGrid >
+	) => {
 		props.setAttributes( {
-			displayLayout: {
-				...props.displayLayout,
-				shrinkColumns: value,
+			templateLayout: {
+				type: LayoutOptions.GRID,
+				...newTemplateLayout,
 			},
 		} );
 	};
 
-	const onPanelDeselect = () => {
+	// Keep track of the last layout values so that we can switch back and forth
+	// without needing to serialize the previous value to the block.
+	const [ lastLayout, setLastLayout ] = useState(
+		autoColumns
+			? DEFAULT_LAYOUT_GRID_COLUMNS
+			: DEFAULT_LAYOUT_GRID_MINIMUM_COLUMN_WIDTH
+	);
+
+	const onAutoColumnsChange = ( value: boolean ) => {
```

Comment:
Nit: The prefix handle is a common convention for event handlers, making it immediately clear that this function is meant to respond to an event.

```suggestion
	const handleAutoColumnsChange = ( value: boolean ) => {
```

Reply:
I _agree_ that traditional convention is for event dispatchers to be prefixed `on` and handlers to be prefixed `handle`. I took a look at our codebase and `/on[^ ]+=\{ on/` is used significantly more than `/on[^ ]+=\{ handle/`. The same difference in frequency can be found in the Gutenberg repository. 

In the interest of consistency, I think it makes the most sense to keep using `on` as the prefix. What do you think?

Reply:
Oh! That makes sense. Thanks for sharing the rational behind it. Really appreciate it 🙌🏻